### PR TITLE
The long awaited 'basename' function

### PR DESCRIPTION
**Pull request by cjllanwarne**
Documentation to follow in Cromwell. Allows:
```
File f = "/Users/chris/blah.txt"
String base = basename(f) # Giving "blah.txt"
String stripped = basename(f, ".txt") # Giving "blah"
```

### CODE REVIEW

**Comment by geoffjentry** [comment]
Can you point the 83737 issues on Cromwell requesting this functionality over here?

**Comment by ruchim** [comment]
:+1: This is awesome! You mentioned documentation will be forthcoming in Cromwell. Would you mind updating the docs in the WDL repo as well?

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 1]
```diff
@@ -329,6 +344,11 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
   }
   def sub(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlStringType)
   def range(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlArrayType(WdlIntegerType))
+  def basename(params: Seq[Try[WdlValue]]): Try[WdlType] = params.toList match {
+    case Success(f) :: Nil if WdlStringType.isCoerceableFrom(f.wdlType) => Success(WdlStringType)
+    case Success(f) :: Success(s) :: Nil if WdlStringType.isCoerceableFrom(f.wdlType) && WdlStringType.isCoerceableFrom(s.wdlType) => Success(WdlStringType)
```
Comment (Horneth) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: Can we use this matching in the implementation above too to get `firstArgument` and `secondArgument` ?

**Comment by Horneth** [comment]
👍

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 2]
```diff
@@ -329,6 +352,11 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
   }
   def sub(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlStringType)
   def range(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlArrayType(WdlIntegerType))
+  def basename(params: Seq[Try[WdlValue]]): Try[WdlType] = params.toList match {
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: Blunder! Should be `WdlType` not `WdlValue`

**Comment by vdauwera** [comment]
Documented in WDL docs here: https://software.broadinstitute.org/wdl/documentation/article?id=10554

**Pull request merged**